### ISSUE TITLE
Bug - CLI - Combination of parameters results in jumbled layer order.

### ISSUE THREAD

**Issue by johanwendin** [aseprite/aseprite#1644]
Hello.

While using the CLI, if you combine the options: 
"--ignore-layer" (to ignore a specific layer, or multiple layers) as well as "--split-layers", the resulting json has the "--list-layers" layer-list order randomized.

Same thing happens if you combine "--split-layers" and specify which layers to include using multiple "--layer"

This also means that it's just not the "layers" meta block that's out-of-order, but the frames list them in the same order as the "layers" meta block, resulting in incorrect sprites if you render them in the order specified within the json.

i.e:

```shell
aseprite -b --split-layers --list-layers filename.ase --format json-array
```

gives: (correct order)
```javascript
"layers": [
 { "name": "_bg", "opacity": 255, "blendMode": "normal" },
 { "name": "legs", "opacity": 255, "blendMode": "normal" },
 { "name": "back_arm", "opacity": 255, "blendMode": "normal" },
 { "name": "body", "opacity": 255, "blendMode": "normal" },
 { "name": "front_arm", "opacity": 255, "blendMode": "normal" },
 { "name": "head", "opacity": 255, "blendMode": "normal" }
]
```

whereas:
```aseprite -b --split-layers --list-layers --ignore-layer _bg filename.ase --format json-array```

gives: (examples, this will vary randomly)

*(first run)*
```javascript
"layers": [
 { "name": "head", "opacity": 255, "blendMode": "normal" },
 { "name": "legs", "opacity": 255, "blendMode": "normal" },
 { "name": "front_arm", "opacity": 255, "blendMode": "normal" },
 { "name": "back_arm", "opacity": 255, "blendMode": "normal" },
 { "name": "body", "opacity": 255, "blendMode": "normal" }
]
```

*(second run)*
```javascript
"layers": [
 { "name": "front_arm", "opacity": 255, "blendMode": "normal" },
 { "name": "legs", "opacity": 255, "blendMode": "normal" },
 { "name": "back_arm", "opacity": 255, "blendMode": "normal" },
 { "name": "head", "opacity": 255, "blendMode": "normal" },
 { "name": "body", "opacity": 255, "blendMode": "normal" }
]
```

*(third run)*
```javascript
"layers": [
 { "name": "back_arm", "opacity": 255, "blendMode": "normal" },
 { "name": "legs", "opacity": 255, "blendMode": "normal" },
 { "name": "head", "opacity": 255, "blendMode": "normal" },
 { "name": "body", "opacity": 255, "blendMode": "normal" },
 { "name": "front_arm", "opacity": 255, "blendMode": "normal" }
]
```

**Expected output:**
```javascript
"layers": [
 { "name": "legs", "opacity": 255, "blendMode": "normal" },
 { "name": "back_arm", "opacity": 255, "blendMode": "normal" },
 { "name": "body", "opacity": 255, "blendMode": "normal" },
 { "name": "front_arm", "opacity": 255, "blendMode": "normal" },
 { "name": "head", "opacity": 255, "blendMode": "normal" }
]
```

### Aseprite and System version

* Aseprite version: 1.2.6-x64, Steam
* System: Windows

**Comment by Brailoff** [comment]
пн, 12 ноя 2018 г., 15:14 David Capello notifications@github.com:

**Comment by vovchisko** [comment]
Can be also reproduced with
``aseprite -b --split-layers --all-layers --ignore-layer ignored-layer-name player.f.ase --sheet player.f.gif``

Still, messing layers order. 1st become 2nd, and the last one becomes first.

It works if I remove ``--ignore-layers``, but w/o ``--ignore-layer`` parameter order is reversed. not sure if it's right behavior.

**Comment by dacap** [comment]
@vovchisko are you using Aseprite v1.2.9? This should be fixed in v1.2.10-beta4
